examples/main/main.cpp

```diff
@@ -475,6 +475,30 @@ bool output_json(struct whisper_context * ctx, const char * fname, const whisper
                 const char * text = whisper_full_get_segment_text(ctx, i);
                 const int64_t t0 = whisper_full_get_segment_t0(ctx, i);
                 const int64_t t1 = whisper_full_get_segment_t1(ctx, i);
+                const char * speaker = "";
+
+                if (params.diarize && pcmf32s.size() == 2) {
+                    const int64_t n_samples = pcmf32s[0].size();
+
+                    const int64_t is0 = timestamp_to_sample(t0, n_samples);
+                    const int64_t is1 = timestamp_to_sample(t1, n_samples);
+
+                    double energy0 = 0.0f;
+                    double energy1 = 0.0f;
+
+                    for (int64_t j = is0; j < is1; j++) {
+                        energy0 += fabs(pcmf32s[0][j]);
+                        energy1 += fabs(pcmf32s[1][j]);
+                    }
+
+                    if (energy0 > 1.1*energy1) {
+                        speaker = "CUSTOMER";
+                    } else if (energy1 > 1.1*energy0) {
+                        speaker = "AGENT";
```

Comment:
Maybe we should leave it as speaker0 and speaker1? And get the change merged upstream as well?

Reply:
I don't want this PR to be merged upstream 😅 

It's hardcoded until we get official support